packages/astro/src/core/build/static-build.ts

```diff
@@ -162,7 +162,7 @@ async function ssrBuild(opts: StaticBuildOptions, internals: BuildInternals, inp
 		root: viteConfig.root,
 		envPrefix: 'PUBLIC_',
 		server: viteConfig.server,
-		base: astroConfig.site ? new URL(astroConfig.site).pathname : '/',
+		base: astroConfig.base,
```

Comment:
This gives Vite the base URL (or the "/" default), making sure that any imported assets include the subpath

i.e. `import image from "../images/penguin.png"` will resolve to `/subpath/assets/asset.123.png`